horovod/tensorflow/mpi_ops.py

```diff
@@ -119,4 +149,15 @@ def broadcast(tensor, root_rank, name=None):
     return MPI_LIB.horovod_broadcast(tensor, name=name, root_rank=root_rank)
 
 
-ops.NotDifferentiable('HorovodBroadcast')
+@ops.RegisterGradient('HorovodBroadcast')
+def broadcast_grad(op, grad):
+    """Gradient for broadcast op.
+
+    Args:
+      op: An operation.
+      grad: `Tensor` gradient with respect to the output of the op.
+
+    Returns:
+      The gradient with respect to the input of the op.
+    """
+    return _allreduce(grad)
```

Comment:
Since broadcast sends data from rank 0 and to everyone else, presumably gradients on ranks > 0 should be always zero?

Reply:
Yes, I think that makes sense.  Please see the updated _broadcast_grad function.